test/test_alias.cpp

```diff
@@ -537,6 +540,28 @@ TEST_F(AliasTest, DuplicateOutputsSegmentedFusion) {
       __FILE__);
 }
 
+namespace {
+
+const FusionExecutor& mostRecentExecutor(const FusionExecutorCache& fec) {
+  const std::vector<FusionExecutor>& executors =
+      fec.getMostRecentKernelRuntime()->executors();
+  return executors.back();
+}
+
+void expectKernelDoesNotStoreToOutput(
+    const FusionExecutor& executor,
+    const int64_t out_index) {
+  // Get the variable name from the `kir::Kernel` not the input fusion, because
+  // they are not always the same.
+  std::string var_name =
+      ir_utils::varName(executor.kernel()->outputs()[out_index]);
+  EXPECT_THAT(
+      executor.kernelString(), Not(ContainsRegex(R"(\b)" + var_name + R"(\[)")))
+      << "The generated CUDA kernel shouldn't store data to `" << var_name
```

Comment:
nitpick: here we are really checking that the variable is not "used" at all. the error message should be consistent, otherwise, in an unlikely event, this will be very confusing.